src/utils/tx-block.ts

```diff
@@ -180,20 +176,20 @@ export class TxBlock {
    * @param args
    * @param type 'address' | 'bool' | 'u8' | 'u16' | 'u32' | 'u64' | 'u128' | 'u256' | 'object'
    */
-  makeMoveVec(args: SuiTxArg[], type?: SuiInputTypes) {
+  makeMoveVec(args: SuiTxArg[], type: SuiInputTypes = getDefaultSuiInputType(args[0])) {
```

Comment:
adding the default argument is probably wrong here.  This should happen after the checks below so the correct error gets thrown